weather/config.py

```diff
@@ -0,0 +1,62 @@
+"""
+Configuration handling utilities.
+"""
+
+import logging
+import os
+from pathlib import Path
+
+import yaml
+
+logger = logging.getLogger(__name__)
+
+
+def load_config(config_path="settings.yaml"):
+    """
+    Load configuration from YAML file and resolve paths.
+
+    Args:
+        config_path: Path to the YAML configuration file
+
+    Returns:
+        dict: Configuration with resolved paths
+    """
+    if not os.path.exists(config_path):
+        logger.warning(
+            f"Configuration file {config_path} not found. Using default values."
+        )
+        return {
+            "ROOT_DIR": str(Path.cwd()),
+            "input_dir": str(Path.cwd() / "input"),
+            "file_name_base": "westfalia_2024_06_01-2024_06_07",
+            "batch_size": 1000,
+            "log_file": None,
+        }
+
+    with open(config_path, "r") as file:
+        config = yaml.safe_load(file)
+
+    # Resolve the ROOT_DIR if it's relative
+    if "ROOT_DIR" in config:
+        root_dir = Path(config["ROOT_DIR"]).expanduser().resolve()
+        config["ROOT_DIR"] = str(root_dir)
```

Comment:
Why would ROOT_DIR ever be in the config? I wouldn't overcomplicate it. Just use the project root dir as retrieved below. The user wouldn't not to configure ROOT_DIR anyways as this is not written anywhere or in the current example config